src/allmydata/test/common_tweb.py

```diff
@@ -0,0 +1,54 @@
+from zope.interface import (
+    classImplements,
+)
+from twisted.python.reflect import (
+    fullyQualifiedName,
+)
+from twisted.internet.defer import (
+    succeed,
+)
+from twisted.web.test.requesthelper import (
+    DummyRequest,
+)
+from twisted.web.iweb import (
+    IRequest,
+)
+from twisted.web.server import (
+    NOT_DONE_YET,
+)
+
+classImplements(DummyRequest, IRequest)
+
+def render(resource, query_args):
+    """
+    Render (in the manner of the Twisted Web Site) a Twisted ``Resource``
+    against a request with the given query arguments .
+
+    :param resource: The page or resource to render.
+
+    :param query_args: The query arguments to put into the request being
+        rendered.  A mapping from ``bytes`` to ``list`` of ``bytes``.
+
+    :return Deferred: A Deferred that fires with the rendered response body as
+        ``bytes``.
+    """
+    request = DummyRequest([])
```

Comment:
Great point.  I used `DummyRequest` because it's easy to fit into the tests and I had the impression `Request` drags in a lot of other baggage that makes using it much more complicated.  However, looks like that's not really the case.  So I've switched this to Request.